Fix proc perv chipunit alignment

deepakala-k opened the pull request:
The pervasive chipunits under the proc got misaligned because the mrw name of that is a substring of odyssey perv chip unit's and so the condition got satisfied.

Make the condition stricter to match the complete string. With this the issue is resolved.

deepakala-k committed:
Fix proc perv chipunit alignment

The pervasive chipunits under the proc got misaligned because the mrw name of
that is a substring of odyssey perv chip unit's and so the condition got
satisfied.

Make the condition stricter to match the complete string. With this the issue is
resolved.

Test results:
pu.perv        k0:n0:s0:p00:c2    0x9C201C0000000000
pu.perv        k0:n0:s0:p00:c3    0x9C201C0000000000
pu.perv        k0:n0:s0:p00:c8    0x8020000100000000
pu.perv        k0:n0:s0:p00:c9    0x8020000100000000
pu.perv        k0:n0:s0:p00:c12   0xA020000100000000
pu.perv        k0:n0:s0:p00:c13   0xA020000100000000
pu.perv        k0:n0:s0:p00:c16   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c17   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c18   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c19   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c24   0xA020000100000000
pu.perv        k0:n0:s0:p00:c26   0xA020000100000000
pu.perv        k0:n0:s0:p00:c27   0xA020000100000000
pu.perv        k0:n0:s0:p00:c29   0xA020000100000000
pu.perv        k0:n0:s0:p00:c31   0xA020000100000000
pu.perv        k0:n0:s0:p00:c32   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c33   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c34   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c35   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c36   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c37   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c38   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c39   0xBC20000000000000
/usr/bin/edbg getscom pu.perv 000F0040 -p0 -call

Device tree dts file after the changes
both of them have FAPI details

perv12 {
compatible = "ibm,power10-chiplet";
index = <0x0c>;
ATTR_CHIPLET_ID = [ 0C ];
ATTR_CHIP_UNIT_POS = [ 0C ];
ATTR_FAPI_NAME = "pu.perv:k0:n0:s0:p00:c12
";
ATTR_FAPI_POS = <12>;
ATTR_HWAS_STATE = [ 00 00 00 00 00 ];
ATTR_MRU_ID = <0x00>;
ATTR_PHYS_BIN_PATH = [ 24 01 00 02 00 05 00 2C 0C 00 00 00 00 00 00 00 00 00 00
00 00  ];
ATTR_PHYS_DEV_PATH = "physical:sys-0/node-0/proc-0/perv-12
";
ATTR_REL_POS = [ 0C ];
ATTR_TYPE = [ 2C ];

...
perv1 {
compatible = "ibm,power10-chiplet";
index = <0x01>;
ATTR_CHIPLET_ID = [ 01 ];
ATTR_CHIP_UNIT_POS = [ 01 ];
ATTR_FAPI_NAME = "ocmb.perv:k0:n0:s0:p00:c1
";
ATTR_FAPI_POS = <32769>;
ATTR_HWAS_STATE = [ 00 00 00 00 00 ];
ATTR_MRU_ID = <0x00>;
ATTR_PHYS_BIN_PATH = [ 24 01 00 02 00 4B 38 2C 01 00 00 00 00 00 00 00 00 00 00
00 00  ];
ATTR_PHYS_DEV_PATH = "physical:sys-0/node-0/ocmb_chip-56/perv-1
";
ATTR_REL_POS = [ 01 ];
ATTR_TYPE = [ 2C ];
};

...

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

dcrowell77 commented on data/p10/FAPITargetsNameMapList.lsv:
```diff
@@ -9,7 +9,7 @@ TARGET_TYPE_PROC_CHIP:chip-processor-power10
 TARGET_TYPE_FC:unit-fc-power10
 TARGET_TYPE_CORE:unit-core-power10
-TARGET_TYPE_PERV:unit-perv-power10
+TARGET_TYPE_PERV:unit-perv,unit-perv-power10
```
This change looks odd, why the double "unit-perv"?

dcrowell77 reviewed (approved):
Seems okay, would need to see the output to know for sure.

deepakala-k commented:
> Seems okay, would need to see the output to know for sure.

pu.perv        k0:n0:s0:p00:c2    0x9C201C0000000000
pu.perv        k0:n0:s0:p00:c3    0x9C201C0000000000
pu.perv        k0:n0:s0:p00:c8    0x8020000100000000
pu.perv        k0:n0:s0:p00:c9    0x8020000100000000
pu.perv        k0:n0:s0:p00:c12   0xA020000100000000
pu.perv        k0:n0:s0:p00:c13   0xA020000100000000
pu.perv        k0:n0:s0:p00:c16   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c17   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c18   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c19   0x9C20000000000000
pu.perv        k0:n0:s0:p00:c24   0xA020000100000000
pu.perv        k0:n0:s0:p00:c26   0xA020000100000000
pu.perv        k0:n0:s0:p00:c27   0xA020000100000000
pu.perv        k0:n0:s0:p00:c29   0xA020000100000000
pu.perv        k0:n0:s0:p00:c31   0xA020000100000000
pu.perv        k0:n0:s0:p00:c32   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c33   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c34   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c35   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c36   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c37   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c38   0xBC20000000000000
pu.perv        k0:n0:s0:p00:c39   0xBC20000000000000
/usr/bin/edbg getscom pu.perv 000F0040 -p0 -call

Device tree dts file after the changes
both of them have FAPI details

perv12 {
compatible = "ibm,power10-chiplet";
index = <0x0c>;
ATTR_CHIPLET_ID = [ 0C ];
ATTR_CHIP_UNIT_POS = [ 0C ];
ATTR_FAPI_NAME = "pu.perv:k0:n0:s0:p00:c12
";
ATTR_FAPI_POS = <12>;
ATTR_HWAS_STATE = [ 00 00 00 00 00 ];
ATTR_MRU_ID = <0x00>;
ATTR_PHYS_BIN_PATH = [ 24 01 00 02 00 05 00 2C 0C 00 00 00 00 00 00 00 00 00 00
00 00  ];
ATTR_PHYS_DEV_PATH = "physical:sys-0/node-0/proc-0/perv-12
";
ATTR_REL_POS = [ 0C ];
ATTR_TYPE = [ 2C ];

...
perv1 {
compatible = "ibm,power10-chiplet";
index = <0x01>;
ATTR_CHIPLET_ID = [ 01 ];
ATTR_CHIP_UNIT_POS = [ 01 ];
ATTR_FAPI_NAME = "ocmb.perv:k0:n0:s0:p00:c1
";
ATTR_FAPI_POS = <32769>;
ATTR_HWAS_STATE = [ 00 00 00 00 00 ];
ATTR_MRU_ID = <0x00>;
ATTR_PHYS_BIN_PATH = [ 24 01 00 02 00 4B 38 2C 01 00 00 00 00 00 00 00 00 00 00
00 00  ];
ATTR_PHYS_DEV_PATH = "physical:sys-0/node-0/ocmb_chip-56/perv-1
";
ATTR_REL_POS = [ 01 ];
ATTR_TYPE = [ 2C ];
};

deepakala-k committed:
Attributes export: Fix the DIMMs, Memport numbering

The attributes tool uses the index value in addition with Proc number to create
the cronus name. And if the index were relative numbering with the ocmb, we
found the numbers are repeated. So using the FAPI position as the index.

Test output:

root@p10bmc:/tmp# attributes export | grep "target\|ATTR_HWAS_STATE\|physical" |
awk '/target = p10.dimm:/{nr[NR]; nr[NR+1];nr[NR+2]}; NR in nr' | awk
'ORS=NR%3?" ; ":"\n"' | grep -w 0xe0
target = p10.dimm:k0:n0:s0:p00:c8 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0 ;
ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-116"
target = p10.dimm:k0:n0:s0:p00:c10 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0 ;
ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-117"
target = p10.dimm:k0:n0:s0:p00:c12 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0 ;
ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-124"
target = p10.dimm:k0:n0:s0:p00:c14 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0 ;
ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-125"
target = p10.dimm:k0:n0:s0:p02:c136 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0
; ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-4"
target = p10.dimm:k0:n0:s0:p02:c138 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0
; ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-5"
target = p10.dimm:k0:n0:s0:p02:c140 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0
; ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-12"
target = p10.dimm:k0:n0:s0:p02:c142 ; ATTR_HWAS_STATE    cpx    0x00000000 0xe0
; ATTR_PHYS_DEV_PATH    str    "physical:sys-0/node-0/dimm-13"

root@ever6bmc:/tmp# attributes export | grep "target\|ATTR_HWAS_STATE\|physical"
| awk '/target = p10.mem_port:/{nr[NR]; nr[NR+1];nr[NR+2]}; NR in nr' | awk
'ORS=NR%3?" ; ":"\n"' | grep -w 0xe0
target = p10.mem_port:k0:n0:s0:p00:c4 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-58/mem_port-0"
target = p10.mem_port:k0:n0:s0:p00:c5 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-58/mem_port-1"
target = p10.mem_port:k0:n0:s0:p00:c6 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-62/mem_port-0"
target = p10.mem_port:k0:n0:s0:p00:c7 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-62/mem_port-1"
target = p10.mem_port:k0:n0:s0:p02:c68 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-2/mem_port-0"
target = p10.mem_port:k0:n0:s0:p02:c69 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-2/mem_port-1"
target = p10.mem_port:k0:n0:s0:p02:c70 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-6/mem_port-0"
target = p10.mem_port:k0:n0:s0:p02:c71 ; ATTR_HWAS_STATE    cpx    0x00000000
0xe0 ; ATTR_PHYS_DEV_PATH    str
"physical:sys-0/node-0/ocmb_chip-6/mem_port-1"

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

devenrao commented:
approved

Swarnendu-R-C reviewed (approved):
Approved
